pkg/kubelet/kubeletconfig/util/files/files.go

```diff
@@ -24,7 +24,10 @@ import (
 	utilfs "k8s.io/kubernetes/pkg/util/filesystem"
 )
 
-const defaultPerm = 0666
+const (
+	defaultPerm = 0755
```

Comment:
can you add a brief comment explaining the permissions?  Why do we want it to be executable, and what is it the permission for?

Reply:
If you don't make a directory executable, you can't access the items in it. I thought about doing things differently for files vs. dirs, but decided to keep it uniform for simplicity.

With directories, AFAIK:
- `r`: allows listing contained files (well, sort of, you'll see the file names but get other permission errors from `ls` if this is all you have, since you need `x` to read the permission bits, size, timestamps, etc. for the files).
- `w`: allows creating files (in practice you often also need `x` to effectively create files, e.g `touch` fails without `x`, I don't actually know if there's a scenario where `w` is useful on its own)
- `x`: allows accessing files (e.g. `cat`) and changing into the directory (e.g. `cd`)

Reply:
good to know.  Was mostly just curious.